pkg/network/master/egressip.go

```diff
@@ -163,6 +165,18 @@ func (eim *egressIPManager) poll(stop chan struct{}) {
 	}
 }
 
+func isNodeConditionGood(cond v1.NodeCondition) bool {
+	isNodeGood := true
+	if cond.Type == v1.NodeReady {
+		klog.Warningf("Node condition type is %s", cond.Type)
+		if cond.Status == v1.ConditionFalse || cond.Status == v1.ConditionUnknown {
+			klog.Warningf("Node condition status is %s", cond.Status)
+			isNodeGood = false
+		}
+	}
+	return isNodeGood
+}
```

Comment:
This is a confusing function. I'd change it to `nodeIsReady(node *v1.Node)`, and move the `range node.Status.Conditions` loop into here. Then you don't have the weirdness of claiming that any non-`NodeReady` condition counts as "good"

Reply:
I've refactored it, will test on a cluster in any case.